src/neuralnet/researcher.cpp

```diff
@@ -119,19 +121,15 @@ std::vector<std::string> FetchProjectsXml()
 
     // Drop the first element which never contains a project:
     //
-    // We could swap-n-pop the element to avoid shifting the whole sequence.
-    // However, BOINC sorts the projects in client_state.xml by name, and we
-    // select the last valid CPID present in this file. To avoid a surprise,
-    // we'll just erase the first item. This routine doesn't run very often.
-    //
-    projects.erase(projects.begin());
+    std::swap(projects.front(), projects.back());
+    projects.pop_back();
```

Comment:
Switch to `std::map` as the backing-storage for `NN::MiningProjectMap` lets us use this optimization now.